src/main/java/TaskOrganizer.java

```diff
@@ -0,0 +1,132 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+
+public class TaskOrganizer {
+    /** Unique ID to be given to each new task for identification */
+    private int newTaskID;
+
+    /** A container containing all tasks added */
+    private HashMap<Integer, Task> tasks;
+
+    /** A list containing all valid task types */
+    private ArrayList<String> taskTypes;
+
+    public TaskOrganizer() {
+        newTaskID = 1;
+        tasks = new HashMap<Integer, Task>();
+        taskTypes = new ArrayList<String>(
+                Arrays.asList("todo", "event", "deadline")
+        );
+    }
+
+    /**
+     * Checks if the type given is a valid task type.
+     *
+     * @param type The type to be checked.
+     * @return True if type is a valid task type, False if type is not a valid task type.
+     */
+    public boolean isTaskType(String type) {
+        return taskTypes.contains(type);
+    }
+
+    /**
+     * Checks if the index provides is out of bounds from the current available tasks.
+     *
+     * @param index The index provided.
+     * @return True if the index provided is out of bounds, False if the index provides is not out of bounds.
+     */
+    public boolean isOutOfBounds(int index) {
+        return (index <= 0 || index >= newTaskID);
+    }
+
+    /**
+     * Checks if tasks is empty.
+     *
+     * @return True if tasks is empty, False if tasks is not empty.
+     */
+    public boolean isEmpty() {
+        return tasks.isEmpty();
+    }
+
+    /**
+     * Adds a task to the list of tasks.
+     *
+     * @param taskType The type of the task to be added.
+     * @param taskName The name of the task to be added.
+     * @param taskDate A string containing the dates of the task (E.g deadline or start and end date).
+     * @return True as task have been added without any errors, False if there are errors when adding the task.
+     */
+    public boolean addTask(String taskType, String taskName, String taskDate) {
```

Comment:
Perhaps there could be a better naming here for addTask to sound like a boolean function